packages/builder/src/schemas.zod.ts

```diff
@@ -24,6 +24,21 @@ const jsonAbiPathRegex = RegExp(/^(?!.*\.d?$).*\.json?$/, 'i');
 const artifactNameRegex = RegExp(/^[A-Z]{1}[\w]+$/, 'i');
 const artifactPathRegex = RegExp(/^.*\.sol:\w+/, 'i');
 
+// Invoke target string schema
+const targetString = z.string().refine(
+  (val) =>
+    ethers.utils.isAddress(val) ||
+    !!val.match(interpolatedRegex) ||
+    !!val.match(stepRegex) ||
+    !!val.match(artifactNameRegex) ||
+    !!val.match(artifactPathRegex),
+  (val) => ({
+    message: `"${val}" must be a valid ethereum address, previously defined contract step name, contract artifact name or filepath`,
```

Comment:
`previously defined contract step name` => `existing contract step name`

its not necessary that the contract step was "previous" (although it must be depended upon) (soon we are  getting rid of `depends` so probably just best to keep response simple)